notebooks/109-performance-tricks/109-latency-tricks.ipynb

```diff
@@ -0,0 +1,407 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import sys\n",
+    "import time\n",
+    "from pathlib import Path\n",
+    "\n",
+    "sys.path.append(\"../utils\")\n",
+    "import notebook_utils as utils"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data\n",
+    "\n",
+    "For all experiments below we're using the same image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import cv2\n",
+    "\n",
+    "IMAGE_WIDTH = 640\n",
+    "IMAGE_HEIGHT = 640\n",
+    "\n",
+    "image = utils.load_image(\"../data/image/coco_bike.jpg\")\n",
+    "input_image = cv2.resize(image, dsize=(IMAGE_WIDTH, IMAGE_HEIGHT), interpolation=cv2.INTER_AREA)\n",
+    "input_image = np.expand_dims(np.transpose(input_image, axes=(2, 0, 1)), axis=0)\n",
+    "utils.show_array(image)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model\n",
+    "\n",
+    "The model we selected is for object detection."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "\n",
+    "base_model_dir = Path(\"model\")\n",
+    "model_name = \"yolov5n\"\n",
+    "model_path = base_model_dir / model_name\n",
+    "\n",
+    "pytorch_model = torch.hub.load(\"ultralytics/yolov5\", \"custom\", path=model_path, device='cpu')\n",
+    "pytorch_model.eval()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Hardware\n",
+    "\n",
+    "The code below lists the available hardware. The hardware below is used in the benchmarking process."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openvino.runtime as ov\n",
+    "\n",
+    "core = ov.Core()\n",
+    "\n",
+    "for device in core.available_devices:\n",
+    "    device_name = core.get_property(device, \"FULL_DEVICE_NAME\")\n",
+    "    print(f\"{device}: {device_name}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Optimizations\n",
+    "\n",
+    "We're defining a benchmark model function to use it for all optimized models below. It runs inference 100 times and average the time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# todo make it 1000\n",
+    "INFER_NUMBER = 100\n",
+    "\n",
+    "def benchmark_model(model, input, model_name, device=\"CPU\"):\n",
```

Comment:
My understanding is this method is used to calculate the latency in Sync mode, but not for latency when runtime is in Async mode. Because in Async mode, there are many parallel tasks, and we only record the total time, not each's task's time.

Reply:
You're absolutely right. The latency will be the same.